MACSio.spec

```diff
@@ -9,6 +9,7 @@ Source0:        https://github.com/LLNL/%{name}/archive/v%{version}.tar.gz
 
 %if 0%{?suse_version}
 BuildRequires:  gcc-fortran
+BuildRequires:  Modules
```

Comment:
We're actually standardising on _lua-lmod_.  There is a PR for adding it to our repos in daos-stack/lua-lmod#1. I have not landed it yet as it's still being validated with the bigger body of work that is aimed at running Leap 15 in CI.  You can use `PR-repos: lua-lmod@PR-1` while it's still in a PR-only state.

This all could revert back to _Modules_ at some point in the future, if/when somebody has time to make it work up and down the entire CI stack, but I had to give up on that as it was becoming a time-suck and building a _lua-lmod_ was just the easier path.

Reply:
I ran into this problem after applying [packaging(#30)](https://github.com/daos-stack/packaging/pull/30) and using `PR-repos: lua-lmod@PR-1`:
```
[2020-06-23T13:39:02.965Z] hdf5:PR-6:lastSuccessfulBuild                    82 kB/s |  13 kB     00:00    
[2020-06-23T13:39:02.965Z] lua-lmod:PR-1:lastSuccessfulBuild                11 kB/s | 546  B     00:00    
[2020-06-23T13:39:02.965Z] Failed to download metadata for repo 'lua-lmod:PR-1:lastSuccessfulBuild'
[2020-06-23T13:39:02.965Z] Error: Failed to download metadata for repo 'lua-lmod:PR-1:lastSuccessfulBuild'
```

Reply:
As a quick test I ran [build 16](https://build.hpdd.intel.com/job/daos-stack/job/MACSio/job/PR-1/16/) without `PR-repos: lua-lmod@PR-1` and re-enabled the `BuildRequires:  Modules` entry in the  spec file.  This confirmed the packaging changes work.

Regardless of Module vs. lua-lmod, I still have a problem with the Leap 15 build where it appears to have issue with the `hdf5-devel` provided `H5pubconf.h` which is renamed by the RPM to be `H5pubconf-64.h`.  This renaming is not a problem on the CentOS 7 build, but it appears to break the Leap 15 build:
```
[ 92%] Building CXX object macsio/CMakeFiles/macsio.dir/__/plugins/macsio_hdf5.c.o
/builddir/build/BUILD/MACSio-1.1/plugins/macsio_hdf5.c:50:10: fatal error: H5pubconf.h: No such file or directory
 #include <H5pubconf.h>
          ^~~~~~~~~~~~~
compilation terminated.
```
Is this a common issue with Leap 15 with a known workaround or is it something specific `hdf5-devel` is introducing that requires modify files or adding a link specific to Leap 15?